lib/flight.rb

```diff
@@ -31,6 +31,13 @@
 
 module Flight
   class << self
+    attr_reader :config
+
+    def load_configuration
+      @config = FlightJob::Configuration.load
+      FlightJob::Configuration.log_warnings
+    end
```

Comment:
I was torn about how this should work. Because of the limitations discussed below, `load_configuration` needs to call `log_warnings`.

Now I would expect `load_configuration` to do two things:
* Reload the config each time it is called
* Be called before `Flight.config` is used.

So I don't see `Flight.config` returning `nil` as a major issue. You are right that `FlightConfiguration::DSL::log_warnings` should probably guard against `nil`.

This is likely a moot point however (see below)

Reply:
If reloading the configuration is needed, I'd expect that to be done by a method `reload_configuration`.  I'd generally expect that an application's configuration does not change during its runtime.  If it does, that ought to be done with a method that's more explicit about it.  The only thing I'd expect of `load_configuration` is that it ensures that the configuration is loaded.

Having `config` potentially return `nil` and `load_configuration` return `Flight.config.__sources__` seems quite surprising.

Reply:
I've reverted `load_configuration` to be an alias to `config`.

Instead `config` is now responsible for logging the warnings and caching itself.